src/structs.h

```diff
@@ -104,17 +104,31 @@ struct Node
                (Parent == other.Parent) &&
                (interval == other.interval);
     }
-    auto friend hash_value(const Node& n) -> std::size_t{
+    auto hash() const -> std::size_t{
```

Comment:
since we are not using return type deduction, I suggest we use the old fashion function delaration for code style consistency eg std::size_t hash() const {...}